openIMIS/developer_tools/management/commands/backend_module_creation.py

```diff
@@ -0,0 +1,107 @@
+from git import Repo
+from openIMIS.openimisapps import openimis_apps
+from pathlib import Path
+
+
+class Command(BaseCommand):
+    help = "This command will generate backend module skeleton in one command"
+
+    def add_arguments(self, parser):
+        parser.add_argument('module_name', type=str)
+        parser.add_argument('author', type=str)
+        parser.add_argument('author_email', type=str)
+
+    def handle(self, *args, **options):
+        # get basic data to create new module like module name, current path/directory
+        author = options['author']
+        author_email = options['author_email']
+
+        module_name = options['module_name']
+        repo_name = f"openimis-be-{module_name}_py"
+        base_path = Path(settings.BASE_DIR)
+        modules_directory = base_path.parent.parent
+
+        new_module_directory = f'{modules_directory}/{repo_name}'
+
+        # check if module exists locally
+        if os.path.exists(f'{modules_directory}/{repo_name}/{module_name}'):
+            raise CommandError("Module already exists in workspace")
+
+        self.__create_project_folder(module_name, new_module_directory)
+        repo = Repo.init(new_module_directory)
+        self.__create_skeleton_module(module_name, new_module_directory)
+
+        self.__add_setup_file(new_module_directory, module_name, author, author_email)
+        self.__add_readme_file(new_module_directory, module_name)
+        self.__add_urls_file(new_module_directory, module_name)
+
+        self.__install_module(module_name, new_module_directory)
+        self.__add_module_to_openimis_json(base_path, module_name)
+
+    def __create_project_folder(self, module_name, new_module_directory):
+        """ create empty folder for new project """
+        try:
+            os.mkdir(new_module_directory)
+            os.mkdir(f'{new_module_directory}/{module_name}')
+        except OSError:
+            raise CommandError("Creation of the directory %s failed" % new_module_directory)
+        else:
+            self.stdout.write(self.style.SUCCESS("Successfully created the directory %s " % new_module_directory))
+
+    def __create_skeleton_module(self, module_name, new_module_directory):
+        result = os.system(f'python manage.py startapp {module_name} {new_module_directory}/{module_name}')
+        if result != 0:
+            raise CommandError(f'skeleton module not crated properly, ended with code: {result}')
+        else:
+            self.stdout.write(self.style.SUCCESS('Succesfully created skeleton module'))
+
+    def __add_readme_file(self, new_module_directory, module_name):
+        readme_file = f'{new_module_directory}/README.md'
+        f = open(readme_file, "w+")
+        f.write(get_skeleton_readme(module_name))
+        f.close()
+        self.stdout.write(self.style.SUCCESS('Succesfully created README.md file'))
+
+    def __add_urls_file(self, new_module_directory, module_name):
```

Comment:
Implementation of __add_urls_file, __add_setup_file and __add_readme_file seems very similar. 
Can you make it more generic?

Reply:
ok

Reply:
yes. I can.